precompile/contracts/warp/contract_warp_handler.go

```diff
@@ -61,8 +61,14 @@ func handleWarpMessage(accessibleState contract.AccessibleState, input []byte, s
 	warpIndex := int(warpIndexInput) // This conversion is safe even if int is 32 bits because we checked above.
 	state := accessibleState.GetStateDB()
 	predicateBytes, exists := state.GetPredicateStorageSlots(ContractAddress, warpIndex)
-	predicateResults := accessibleState.GetBlockContext().GetPredicateResults(state.GetTxHash(), ContractAddress)
-	valid := exists && !set.BitsFromBytes(predicateResults).Contains(warpIndex)
+	predicateResultsBytes := accessibleState.GetBlockContext().GetPredicateResultsBytes()
```

Comment:
I wonder if we should avoid parsing this twice and return direct parsed results (and pass the parsed results when we initialiaze a new block context at the first place). seems like this is something we can extend here.

Reply:
Also sorry if I we have already discussed this before :D

Reply:
The reason for this is the import loop it would form between params <-> predicates.
Currently the predicates package imports the params package to learn about the DynamicFeeExtraDataSize.
Let's chat about this on Monday. IMO the current code is too messy but I don't necessarily want to fix it in this PR unless we figure something simple out.